app/core/ui/src/main/kotlin/com/hedvig/android/core/ui/MoleculeViewModel.kt

```diff
@@ -0,0 +1,43 @@
+package com.hedvig.android.core.ui
+
+import androidx.compose.runtime.Composable
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import app.cash.molecule.AndroidUiDispatcher
+import app.cash.molecule.RecompositionMode
+import app.cash.molecule.moleculeFlow
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.WhileSubscribed
+import kotlinx.coroutines.flow.stateIn
+import kotlin.time.Duration.Companion.seconds
+
+abstract class MoleculeViewModel<Event, UiState> : ViewModel() {
```

Comment:
(Wrote this in another place too, but just to not confuse you if you see this first)

This is this https://github.com/cashapp/molecule with some video about it here https://www.youtube.com/watch?v=q9p4ewk-9E4.
I wanted to bring it up in the code camp we got soon, so I thought what perfect way to give it a shot in the app too (I had tried it in sample projects before) so that we got something to reference in the code camp.